systemtests/src/main/java/io/enmasse/systemtest/utils/AddressSpaceUtils.java

```diff
@@ -223,4 +229,59 @@ private static EndpointStatus getEndpointByServiceName(String serviceName, List<
     public boolean isBrokered(AddressSpace addressSpace) {
         return addressSpace.getSpec().getType().equals(AddressSpaceType.BROKERED.toString());
     }
+
+    public static AddressSpace waitForAddressSpaceConnectorsReady(AddressSpace addressSpace) throws Exception {
+        return waitForAddressSpaceConnectorsReady(addressSpace, new TimeoutBudget(5, TimeUnit.MINUTES));
+    }
+
+    public static AddressSpace waitForAddressSpaceConnectorsReady(AddressSpace addressSpace, TimeoutBudget budget) throws Exception {
+        waitForAddressSpaceReady(addressSpace);
+        boolean isReady = false;
+        var client = Kubernetes.getInstance().getAddressSpaceClient(addressSpace.getMetadata().getNamespace());
+
+        String name = addressSpace.getMetadata().getName();
+        AddressSpace clientAddressSpace = addressSpace;
+        while (budget.timeLeft() >= 0 && !isReady) {
+            clientAddressSpace = client.withName(name).get();
+            isReady = areAddressSpaceConnectorsReady(clientAddressSpace);
+            if (!isReady) {
+                Thread.sleep(10000);
+            }
+            log.info("Waiting until connectors of address space: '{}' messages {} will be in ready state", name, getConnectorStatuses(clientAddressSpace));
+        }
+
```

Comment:
Add another line after this loop to ensure we do a check after the full timeout is reached.

 isReady = areAddressSpaceConnectorsReady(clientAddressSpace);